lib/Sema/DerivedConformanceCodable.cpp

```diff
@@ -89,6 +89,12 @@ static CodableConformanceType typeConformsToCodable(TypeChecker &tc,
   // Some generic types need to be introspected to get at their "true" Codable
   // conformance.
   auto canType = target->getCanonicalType();
+  if (auto referenceType = dyn_cast<ReferenceStorageType>(canType)) {
```

Comment:
Ah, my bad. If you don't need a canonical type for anything else, then you can just use `getAs`. If you actually want one for some other reason, you can stick to using `dyn_cast`, but then use the "proxy methods" on the various specializations of CanType to stay in canonical land: `referenceType.getReferentType()` instead of `->`.